sdk/src/main/java/com/mapbox/maps/renderer/egl/EGLCore.kt

```diff
@@ -24,45 +33,56 @@ internal class EGLCore(
 
   internal val eglNoSurface: EGLSurface = EGL10.EGL_NO_SURFACE
 
+  /**
+   * For user convenience we will deliver [RendererSetupErrorListener] on main thread.
+   */
+  private val mainHandler = Handler(Looper.getMainLooper())
+  /**
+   * We require error accumulation as renderer is created when [MapView] is created and
+   * some errors could already be triggered on render thread
+   * before user will call [MapView.addRendererSetupErrorListener] -
+   * in that case accumulated errors will still be delivered to the user.
+   */
+  private val accumulatedRendererErrorList = CopyOnWriteArrayList<RendererError>()
+  private val rendererSetupErrorListenerSet = CopyOnWriteArraySet<RendererSetupErrorListener>()
```

Comment:
nit : add annotations or comments to indicate which thread accesses both?

Reply:
they are both touched from both threads

Reply:
For `accumulatedRendererErrorList` there seem one single place with `isNotEmpty` check made from main is that right?

Reply:
`accumulatedRendererErrorList` is also cleared from main thread and local copy is created from main thread as well